src/test/java/com/redhat/devtools/lsp4ij/mock/MockTextDocumentService.java

```diff
@@ -384,7 +384,27 @@ public void setMockSelectionRanges(List<SelectionRange> mockSelectionRanges) {
 
     @Override
     public CompletableFuture<List<SelectionRange>> selectionRange(SelectionRangeParams params) {
-        return CompletableFuture.completedFuture(mockSelectionRanges);
+        // Find the mock selection ranges that apply to the specified position. This allows us to have a single mock
+        // response that covers multiple positions that might be queried during a given test.
+        List<SelectionRange> applicableMockSelectionRanges = mockSelectionRanges

```

Comment:
This allows us to have a single mock `textDocument/selectionRange` response that is actually the merged responses for what would normally be multiple requests, one per-line.